test/e2e/tests/send-hex-address.spec.js

```diff
@@ -233,39 +222,28 @@ describe('Send ERC20 to a 40 character hexadecimal address', function () {
         dapp: true,
         fixtures: 'connected-state',
         ganacheOptions,
+        smartContract,
         title: this.test.title,
         failOnConsoleError: false,
       },
-      async ({ driver }) => {
+      async ({ driver, contractRegistry }) => {
+        const contractAddress = await contractRegistry.getContractAddress(
+          smartContract,
+        );
         await driver.navigate();
         await driver.fill('#password', 'correct horse battery staple');
         await driver.press('#password', driver.Key.ENTER);
 
         // Create TST
-        await driver.openNewPage('http://127.0.0.1:8080/');
-        await driver.clickElement('#createToken');
-        await driver.waitUntilXWindowHandles(3);
+        await driver.openNewPage(
+          `http://127.0.0.1:8080/?contract=${contractAddress}`,
+        );
+
         let windowHandles = await driver.getAllWindowHandles();
         const extension = windowHandles[0];
-        const dapp = await driver.switchToWindowWithTitle(
-          'E2E Test Dapp',
-          windowHandles,
-        );
-        await driver.switchToWindowWithTitle(
-          'MetaMask Notification',
-          windowHandles,
-        );
-        await driver.clickElement({ text: 'Confirm', tag: 'button' });
-        await driver.waitUntilXWindowHandles(2);
-        await driver.switchToWindow(extension);
-        await driver.clickElement('[data-testid="home__activity-tab"]');
-        await driver.waitForSelector(
-          '.transaction-list__completed-transactions .transaction-list-item:nth-of-type(1)',
-          { timeout: 10000 },
-        );
 
         // Add token
-        await driver.switchToWindow(dapp);
+        await driver.findClickableElement('#deployButton');
```

Comment:
Do we still need to look for the `deployButton`? Should we remove it since we're now pre-seeding the contract?

Reply:
@adonesky1 that's a great observation, so thank you for pointing this out.

The reason for adding this previous step of `findClickableElement` is because I noticed that until DeployButton element is not clickable, the Dapp is not actually ready and clicking on WatchAsset directly without waiting for this makes test fail.

First, I was considering to add a delay, but thought this approach is more deterministic. This same step was also mentioned by @PeterYinusa on another PR and we finally decided to use the `findClickableElement` instead of the delay, but if you have strong opinions on the other direction, let me know.

See the other PR with the comment [here](https://github.com/MetaMask/metamask-extension/pull/15553/files#r945764773)

![deploy-contract](https://user-images.githubusercontent.com/54408225/184648128-028ae5ab-91bf-4067-9516-5c6d4f983eca.gif)

Reply:
ah ok, makes sense! Do you mind adding a comment to explain this?

Reply:
ah ok, makes sense! Do you mind adding a comment to explain this?

Reply:
@adonesky1 added a comment, thank you!!!

Reply:
Thanks @seaona !